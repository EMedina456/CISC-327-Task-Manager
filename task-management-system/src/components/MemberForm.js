// Program Intention: Handle the addition, removal, and management of a member
// Input/Output: Handle the username of the member, and permission of the member if needed
// Run Intention: Run with the entire website

// Import files and dependencies here
import { useState } from 'react'
const MemberForm = ({ title }) => {
  // Handle the memebre and permission of the member, permissions, their permissions not needed when removing
  const [member, setMember] = useState('')
  const [yourPermission, setYourPermission] = useState('')
  const [theirPermission, setTheirPermission] = useState('')

  // Handle the submit of the form, currently only console.log the member and permissions
  const handleSubmit = (e) => {
    console.log('name', member)
    console.log('permission', yourPermission)
    console.log('permission', theirPermission)
  }

  // Member Form Page
  return (
    <div className="flex-1 flex-col p-[2%] justify-center items-center h-full">
      {/* Handle the form of member, title changes based on the argument given*/}
      <h1 className="flex text-3xl font-bold mb-2 lg:text-5xl md:text-5xl">
        {title}
      </h1>
      <div className="border-[#60AB9A] w-60 h-1 border-2" />
      <form onSubmit={handleSubmit}>
        <div className="">
          <label>
            {/* Handle the member name input*/}
            <h1 className="text-2xl font-bold mb-4 mt-4 lg:text-2xl md:text-3xl">
              name
            </h1>
            <input
              type="text"
              id="member_name"
              alt="member_name"
              className="box-border h-8 w-44 p-4 border-4"
              onChange={(e) => setMember(e.target.value)}
            />
          </label>
          {/* Handle your permission input*/}
          <label>
            <h1 className="text-2xl font-bold mb-4 mt-4 lg:text-2xl md:text-3xl">
              your permission
            </h1>
            <input
              type="text"
              id="your-permisson"
              className="box-border h-8 w-44 p-4 border-4"
              onChange={(e) => setYourPermission(e.target.value)}
            />
          </label>
          {/* Handle their permission input, not needed when removing a member*/}
          {title === 'Remove a member' ? null : (
            <label>
              <h1 className="text-2xl font-bold mb-4 mt-4 lg:text-2xl md:text-3xl">
                their permission
              </h1>
              <input
                type="text"
                id="their-permisson"
                className="box-border h-8 w-44 p-4 border-4"
                onChange={(e) => setTheirPermission(e.target.value)}
              />
            </label>
          )}

          <div className="justify-center items-center text-left">
            {/* Handle the submit button*/}
            <button className="text-xl font-bold mt-2 underline decoration-[#0acdff]">
              Submit
            </button>
          </div>
        </div>
      </form>
    </div>
  )
}
export default MemberForm
