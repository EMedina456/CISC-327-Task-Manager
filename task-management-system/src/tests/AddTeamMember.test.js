/** @jest-environment jsdom */
// Program Intention: Implement Testing for the Add Team Member Functionality
// Input/Output: Handle the addition of a member
// Run Intention: Run with the other test cases

// Import files and dependencies here
import { render, screen } from '@testing-library/react'
import React from 'react'
import user from '@testing-library/user-event'
import MemberForm from '../components/MemberForm'
import Home from '../pages/Home'
import renderer from 'react-test-renderer'
import { BrowserRouter } from 'react-router-dom'

// Add Team Member Test
describe('Add Team Members', () => {
  // Create a snapshot of the Home page
  const tree = renderer
    .create(
      <BrowserRouter>
        <Home />
      </BrowserRouter>
    )
    .toJSON()

  // Render the Member Form before each test
  beforeEach(() => {
    // eslint-disable-next-line testing-library/no-render-in-setup
    render(<MemberForm title={'Add a member'} />)
  })

  // Test the addition of a member with valid permissions
  it('Scenario Valid Permissions', async () => {
    // Type in the required test fields
    user.type(
      user.click(
        screen.getByRole('textbox', {
          name: /name/i,
        })
      ),
      'Yves'
    )
    user.type(
      user.click(
        screen.getByRole('textbox', {
          name: /your permission/i,
        })
      ),
      'valid permission'
    )
    user.type(
      user.click(
        screen.getByRole('textbox', {
          name: /their permission/i,
        })
      ),
      'random permission'
    )
    user.click(screen.getByText(/submit/i))
    user.click(
      screen.getByRole('button', {
        name: /submit/i,
      })
    )

    // Compare the snapshot
    expect(tree).toMatchSnapshot()

    // CHECK IF THE USER IS ADDED
  })

  // Test the addition of a member that does not exist
  it('Scenario User Does not Exit', async () => {
    // Type in the required test fields
    user.type(
      user.click(
        screen.getByRole('textbox', {
          name: /name/i,
        })
      ),
      'Invalid user'
    )
    user.type(
      user.click(
        screen.getByRole('textbox', {
          name: /your permission/i,
        })
      ),
      'random permission'
    )
    user.type(
      user.click(
        screen.getByRole('textbox', {
          name: /their permission/i,
        })
      ),
      'random permission'
    )
    user.click(screen.getByText(/submit/i))
    user.click(
      screen.getByRole('button', {
        name: /submit/i,
      })
    )

    // Compare the snapshot
    expect(tree).toMatchSnapshot()

    // CHECK ERROR MESSAGE
  })
  // Test the addition of a member with invalid permissions
  it('Scenario Invalid Permissions', async () => {
    // Type in the required test fields
    user.type(
      user.click(
        screen.getByRole('textbox', {
          name: /name/i,
        })
      ),
      ''
    )
    user.type(
      user.click(
        screen.getByRole('textbox', {
          name: /your permission/i,
        })
      ),
      'invalid permission'
    )
    user.type(
      user.click(
        screen.getByRole('textbox', {
          name: /their permission/i,
        })
      ),
      'random permission'
    )
    user.click(screen.getByText(/submit/i))
    user.click(
      screen.getByRole('button', {
        name: /submit/i,
      })
    )
    // Compare the snapshot
    expect(tree).toMatchSnapshot()

    // CHECK ERROR MESSAGE
  })
})
