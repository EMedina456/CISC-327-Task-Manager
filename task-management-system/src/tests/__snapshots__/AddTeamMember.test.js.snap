// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Add Team Members Scenario Invalid Permissions 1`] = `
<div
  className="bg-[#D9D9D9] min-h-[100vh]"
>
  <header
    className="flex min-h-[15%] bg-[#FF9FB2] items-center"
  >
    <button
      onClick={[Function]}
    >
      <h1
        className="text-6xl text-left m-6 font-extrabold"
      >
        Task-it
      </h1>
    </button>
    <div
      className="flex items-center align-center"
    >
      <button
        className="text-6xl text-right ml-auto mr-6"
        onClick={[Function]}
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 1024 1024"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M482 152h60q8 0 8 8v704q0 8-8 8h-60q-8 0-8-8V160q0-8 8-8Z"
          />
          <path
            d="M192 474h672q8 0 8 8v60q0 8-8 8H160q-8 0-8-8v-60q0-8 8-8Z"
          />
        </svg>
      </button>
    </div>
    <a
      className="text-6xl text-right ml-auto mr-6"
      href="/login"
      onClick={[Function]}
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
        <path
          d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zM7.35 18.5C8.66 17.56 10.26 17 12 17s3.34.56 4.65 1.5c-1.31.94-2.91 1.5-4.65 1.5s-3.34-.56-4.65-1.5zm10.79-1.38a9.947 9.947 0 00-12.28 0A7.957 7.957 0 014 12c0-4.42 3.58-8 8-8s8 3.58 8 8c0 1.95-.7 3.73-1.86 5.12z"
        />
        <path
          d="M12 6c-1.93 0-3.5 1.57-3.5 3.5S10.07 13 12 13s3.5-1.57 3.5-3.5S13.93 6 12 6zm0 5c-.83 0-1.5-.67-1.5-1.5S11.17 8 12 8s1.5.67 1.5 1.5S12.83 11 12 11z"
        />
      </svg>
    </a>
  </header>
  <div
    className="flex min-h-[85%]"
  >
    <div
      className="flex flex-col justify-top p-[2%] items-center space-y-8 w-[20%] bg-[#FBDCE2]"
    >
      <h1
        className="text-4xl font-bold mb-2"
      >
        Sort By 
      </h1>
      <div
        className="border-[#60AB9A] w-full h-1 border-2"
      />
      <div
        className="flex flex-row "
      >
        <div
          className="flex flex-col"
        >
          <button
            alt="handleDeadline"
            id="handleDeadline"
            onClick={[Function]}
          >
            <h1
              className="text-3xl m-2 font-bold mb-2"
            >
              Deadline 
            </h1>
            <div
              className="mr-2 border-[#60AB9A] w-full h-1 border-2"
            />
          </button>
        </div>
        <div
          className="flex flex-col"
        >
          <button
            alt="handlePriority"
            id="handlePriority"
            onClick={[Function]}
          >
            <h1
              className="text-3xl m-2 font-bold mb-2"
            >
              Priority 
            </h1>
            <div
              className=" ml-2 border-[#60AB9A] w-full h-1 border-2"
            />
          </button>
        </div>
      </div>
      <div
        className=" space-y-8 "
      >
        <div
          className="flex flex-col justify-center"
        >
          <h1
            className="text-4xl font-bold mb-2"
          >
            !!! Priority
          </h1>
          <div
            className="border-[#60AB9A] w-full h-1 border-2"
          />
          <button
            className="text-xl text-left font-bold mt-2 underline decoration-[#0acdff]"
            onClick={[Function]}
          >
            Task 1
          </button>
          <button
            className="text-xl text-left font-bold mt-2 underline decoration-[#0acdff]"
            onClick={[Function]}
          >
            Task 2
          </button>
          <button
            className="text-xl text-left font-bold mt-2 underline decoration-[#0acdff]"
            onClick={[Function]}
          >
            Task 3
          </button>
        </div>
        <div
          className="flex flex-col justify-center "
        >
          <h1
            className="text-4xl font-bold mb-2"
          >
            !! Priority
          </h1>
          <div
            className="border-[#60AB9A] w-full h-1 border-2"
          />
          <button
            className="text-xl text-left font-bold mt-2 underline decoration-[#0acdff]"
            onClick={[Function]}
          >
            Task 1
          </button>
          <button
            className="text-xl text-left font-bold mt-2 underline decoration-[#0acdff]"
            onClick={[Function]}
          >
            Task 2
          </button>
          <button
            className="text-xl text-left font-bold mt-2 underline decoration-[#0acdff]"
            onClick={[Function]}
          >
            Task 3
          </button>
        </div>
        <div
          className="flex flex-col justify-center"
        >
          <h1
            className="text-4xl font-bold mb-2"
          >
            ! Priority
          </h1>
          <div
            className="border-[#60AB9A] w-full h-1 border-2"
          />
          <button
            className="text-xl text-left font-bold mt-2 underline decoration-[#0acdff]"
            onClick={[Function]}
          >
            Task 1
          </button>
          <button
            className="text-xl text-left font-bold mt-2 underline decoration-[#0acdff]"
            onClick={[Function]}
          >
            Task 2
          </button>
          <button
            className="text-xl text-left font-bold mt-2 underline decoration-[#0acdff]"
            onClick={[Function]}
          >
            Task 3
          </button>
        </div>
      </div>
    </div>
    <div
      className="flex flex-col md:flex-row w-[80%]"
    >
      <div
        className="flex-1 flex-col p-[2%] justify-center items-center h-full"
      >
        <h1
          className="flex text-5xl font-bold mb-2"
        >
          Projects
        </h1>
        <div
          className="border-[#60AB9A] w-80 h-1 border-2"
        />
        <button
          className="flex text-2xl font-bold mt-2 underline decoration-[#0acdff]"
          onClick={[Function]}
        >
          Project 1
        </button>
        <button
          className="flex text-2xl font-bold mt-2 underline decoration-[#0acdff]"
          onClick={[Function]}
        >
          Project 2
        </button>
        <button
          className="flex text-2xl font-bold mt-2 underline decoration-[#0acdff]"
          onClick={[Function]}
        >
          Project 3
        </button>
        <button
          className="flex text-2xl font-bold mt-2 underline decoration-[#0acdff]"
          onClick={[Function]}
        >
          ...
        </button>
        <div
          className="my-16"
        />
        <h1
          className="flex text-5xl font-bold mb-2"
        >
          Overdue Tasks
        </h1>
        <div
          className="border-[#60AB9A] w-80 h-1 border-2"
        />
        <button
          className="flex text-2xl font-bold mt-2 underline decoration-[#0acdff]"
          onClick={[Function]}
        >
          Task 1
        </button>
        <button
          className="flex text-2xl font-bold mt-2 underline decoration-[#0acdff]"
          onClick={[Function]}
        >
          Task 2
        </button>
        <button
          className="flex text-2xl font-bold mt-2 underline decoration-[#0acdff]"
          onClick={[Function]}
        >
          Task 3
        </button>
        <button
          className="flex text-2xl font-bold mt-2 underline decoration-[#0acdff]"
          onClick={[Function]}
        >
          ...
        </button>
      </div>
      <div
        className="flex-1 flex-col p-[2%] justify-center items-center h-full"
      >
        <h1
          className="flex text-5xl font-bold mb-2"
        >
          Tasks
        </h1>
        <div
          className="border-[#60AB9A] w-80 h-1 border-2"
        />
        <button
          className="flex text-2xl font-bold mt-2 underline decoration-[#0acdff]"
          onClick={[Function]}
        >
          Task 1
        </button>
        <button
          className="flex text-2xl font-bold mt-2 underline decoration-[#0acdff]"
          onClick={[Function]}
        >
          Task 2
        </button>
        <button
          className="flex text-2xl font-bold mt-2 underline decoration-[#0acdff]"
          onClick={[Function]}
        >
          Task 3
        </button>
        <button
          className="flex text-2xl font-bold mt-2 underline decoration-[#0acdff]"
          onClick={[Function]}
        >
          Task 4
        </button>
        <button
          className="flex text-2xl font-bold mt-2 underline decoration-[#0acdff]"
          onClick={[Function]}
        >
          Task 5
        </button>
        <button
          className="flex text-2xl font-bold mt-2 underline decoration-[#0acdff]"
          onClick={[Function]}
        >
          Task 6
        </button>
        <button
          className="flex text-2xl font-bold mt-2 underline decoration-[#0acdff]"
          onClick={[Function]}
        >
          Task 7
        </button>
        <button
          className="flex text-2xl font-bold mt-2 underline decoration-[#0acdff]"
          onClick={[Function]}
        >
          Task 8
        </button>
        <button
          className="flex text-2xl font-bold mt-2 underline decoration-[#0acdff]"
          onClick={[Function]}
        >
          Task 9
        </button>
        <button
          className="flex text-2xl font-bold mt-2 underline decoration-[#0acdff]"
          onClick={[Function]}
        >
          ...
        </button>
      </div>
      <div
        className="my-16"
      />
    </div>
  </div>
</div>
`;

exports[`Add Team Members Scenario User Does not Exit 1`] = `
<div
  className="bg-[#D9D9D9] min-h-[100vh]"
>
  <header
    className="flex min-h-[15%] bg-[#FF9FB2] items-center"
  >
    <button
      onClick={[Function]}
    >
      <h1
        className="text-6xl text-left m-6 font-extrabold"
      >
        Task-it
      </h1>
    </button>
    <div
      className="flex items-center align-center"
    >
      <button
        className="text-6xl text-right ml-auto mr-6"
        onClick={[Function]}
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 1024 1024"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M482 152h60q8 0 8 8v704q0 8-8 8h-60q-8 0-8-8V160q0-8 8-8Z"
          />
          <path
            d="M192 474h672q8 0 8 8v60q0 8-8 8H160q-8 0-8-8v-60q0-8 8-8Z"
          />
        </svg>
      </button>
    </div>
    <a
      className="text-6xl text-right ml-auto mr-6"
      href="/login"
      onClick={[Function]}
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
        <path
          d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zM7.35 18.5C8.66 17.56 10.26 17 12 17s3.34.56 4.65 1.5c-1.31.94-2.91 1.5-4.65 1.5s-3.34-.56-4.65-1.5zm10.79-1.38a9.947 9.947 0 00-12.28 0A7.957 7.957 0 014 12c0-4.42 3.58-8 8-8s8 3.58 8 8c0 1.95-.7 3.73-1.86 5.12z"
        />
        <path
          d="M12 6c-1.93 0-3.5 1.57-3.5 3.5S10.07 13 12 13s3.5-1.57 3.5-3.5S13.93 6 12 6zm0 5c-.83 0-1.5-.67-1.5-1.5S11.17 8 12 8s1.5.67 1.5 1.5S12.83 11 12 11z"
        />
      </svg>
    </a>
  </header>
  <div
    className="flex min-h-[85%]"
  >
    <div
      className="flex flex-col justify-top p-[2%] items-center space-y-8 w-[20%] bg-[#FBDCE2]"
    >
      <h1
        className="text-4xl font-bold mb-2"
      >
        Sort By 
      </h1>
      <div
        className="border-[#60AB9A] w-full h-1 border-2"
      />
      <div
        className="flex flex-row "
      >
        <div
          className="flex flex-col"
        >
          <button
            alt="handleDeadline"
            id="handleDeadline"
            onClick={[Function]}
          >
            <h1
              className="text-3xl m-2 font-bold mb-2"
            >
              Deadline 
            </h1>
            <div
              className="mr-2 border-[#60AB9A] w-full h-1 border-2"
            />
          </button>
        </div>
        <div
          className="flex flex-col"
        >
          <button
            alt="handlePriority"
            id="handlePriority"
            onClick={[Function]}
          >
            <h1
              className="text-3xl m-2 font-bold mb-2"
            >
              Priority 
            </h1>
            <div
              className=" ml-2 border-[#60AB9A] w-full h-1 border-2"
            />
          </button>
        </div>
      </div>
      <div
        className=" space-y-8 "
      >
        <div
          className="flex flex-col justify-center"
        >
          <h1
            className="text-4xl font-bold mb-2"
          >
            !!! Priority
          </h1>
          <div
            className="border-[#60AB9A] w-full h-1 border-2"
          />
          <button
            className="text-xl text-left font-bold mt-2 underline decoration-[#0acdff]"
            onClick={[Function]}
          >
            Task 1
          </button>
          <button
            className="text-xl text-left font-bold mt-2 underline decoration-[#0acdff]"
            onClick={[Function]}
          >
            Task 2
          </button>
          <button
            className="text-xl text-left font-bold mt-2 underline decoration-[#0acdff]"
            onClick={[Function]}
          >
            Task 3
          </button>
        </div>
        <div
          className="flex flex-col justify-center "
        >
          <h1
            className="text-4xl font-bold mb-2"
          >
            !! Priority
          </h1>
          <div
            className="border-[#60AB9A] w-full h-1 border-2"
          />
          <button
            className="text-xl text-left font-bold mt-2 underline decoration-[#0acdff]"
            onClick={[Function]}
          >
            Task 1
          </button>
          <button
            className="text-xl text-left font-bold mt-2 underline decoration-[#0acdff]"
            onClick={[Function]}
          >
            Task 2
          </button>
          <button
            className="text-xl text-left font-bold mt-2 underline decoration-[#0acdff]"
            onClick={[Function]}
          >
            Task 3
          </button>
        </div>
        <div
          className="flex flex-col justify-center"
        >
          <h1
            className="text-4xl font-bold mb-2"
          >
            ! Priority
          </h1>
          <div
            className="border-[#60AB9A] w-full h-1 border-2"
          />
          <button
            className="text-xl text-left font-bold mt-2 underline decoration-[#0acdff]"
            onClick={[Function]}
          >
            Task 1
          </button>
          <button
            className="text-xl text-left font-bold mt-2 underline decoration-[#0acdff]"
            onClick={[Function]}
          >
            Task 2
          </button>
          <button
            className="text-xl text-left font-bold mt-2 underline decoration-[#0acdff]"
            onClick={[Function]}
          >
            Task 3
          </button>
        </div>
      </div>
    </div>
    <div
      className="flex flex-col md:flex-row w-[80%]"
    >
      <div
        className="flex-1 flex-col p-[2%] justify-center items-center h-full"
      >
        <h1
          className="flex text-5xl font-bold mb-2"
        >
          Projects
        </h1>
        <div
          className="border-[#60AB9A] w-80 h-1 border-2"
        />
        <button
          className="flex text-2xl font-bold mt-2 underline decoration-[#0acdff]"
          onClick={[Function]}
        >
          Project 1
        </button>
        <button
          className="flex text-2xl font-bold mt-2 underline decoration-[#0acdff]"
          onClick={[Function]}
        >
          Project 2
        </button>
        <button
          className="flex text-2xl font-bold mt-2 underline decoration-[#0acdff]"
          onClick={[Function]}
        >
          Project 3
        </button>
        <button
          className="flex text-2xl font-bold mt-2 underline decoration-[#0acdff]"
          onClick={[Function]}
        >
          ...
        </button>
        <div
          className="my-16"
        />
        <h1
          className="flex text-5xl font-bold mb-2"
        >
          Overdue Tasks
        </h1>
        <div
          className="border-[#60AB9A] w-80 h-1 border-2"
        />
        <button
          className="flex text-2xl font-bold mt-2 underline decoration-[#0acdff]"
          onClick={[Function]}
        >
          Task 1
        </button>
        <button
          className="flex text-2xl font-bold mt-2 underline decoration-[#0acdff]"
          onClick={[Function]}
        >
          Task 2
        </button>
        <button
          className="flex text-2xl font-bold mt-2 underline decoration-[#0acdff]"
          onClick={[Function]}
        >
          Task 3
        </button>
        <button
          className="flex text-2xl font-bold mt-2 underline decoration-[#0acdff]"
          onClick={[Function]}
        >
          ...
        </button>
      </div>
      <div
        className="flex-1 flex-col p-[2%] justify-center items-center h-full"
      >
        <h1
          className="flex text-5xl font-bold mb-2"
        >
          Tasks
        </h1>
        <div
          className="border-[#60AB9A] w-80 h-1 border-2"
        />
        <button
          className="flex text-2xl font-bold mt-2 underline decoration-[#0acdff]"
          onClick={[Function]}
        >
          Task 1
        </button>
        <button
          className="flex text-2xl font-bold mt-2 underline decoration-[#0acdff]"
          onClick={[Function]}
        >
          Task 2
        </button>
        <button
          className="flex text-2xl font-bold mt-2 underline decoration-[#0acdff]"
          onClick={[Function]}
        >
          Task 3
        </button>
        <button
          className="flex text-2xl font-bold mt-2 underline decoration-[#0acdff]"
          onClick={[Function]}
        >
          Task 4
        </button>
        <button
          className="flex text-2xl font-bold mt-2 underline decoration-[#0acdff]"
          onClick={[Function]}
        >
          Task 5
        </button>
        <button
          className="flex text-2xl font-bold mt-2 underline decoration-[#0acdff]"
          onClick={[Function]}
        >
          Task 6
        </button>
        <button
          className="flex text-2xl font-bold mt-2 underline decoration-[#0acdff]"
          onClick={[Function]}
        >
          Task 7
        </button>
        <button
          className="flex text-2xl font-bold mt-2 underline decoration-[#0acdff]"
          onClick={[Function]}
        >
          Task 8
        </button>
        <button
          className="flex text-2xl font-bold mt-2 underline decoration-[#0acdff]"
          onClick={[Function]}
        >
          Task 9
        </button>
        <button
          className="flex text-2xl font-bold mt-2 underline decoration-[#0acdff]"
          onClick={[Function]}
        >
          ...
        </button>
      </div>
      <div
        className="my-16"
      />
    </div>
  </div>
</div>
`;

exports[`Add Team Members Scenario Valid Permissions 1`] = `
<div
  className="bg-[#D9D9D9] min-h-[100vh]"
>
  <header
    className="flex min-h-[15%] bg-[#FF9FB2] items-center"
  >
    <button
      onClick={[Function]}
    >
      <h1
        className="text-6xl text-left m-6 font-extrabold"
      >
        Task-it
      </h1>
    </button>
    <div
      className="flex items-center align-center"
    >
      <button
        className="text-6xl text-right ml-auto mr-6"
        onClick={[Function]}
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 1024 1024"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M482 152h60q8 0 8 8v704q0 8-8 8h-60q-8 0-8-8V160q0-8 8-8Z"
          />
          <path
            d="M192 474h672q8 0 8 8v60q0 8-8 8H160q-8 0-8-8v-60q0-8 8-8Z"
          />
        </svg>
      </button>
    </div>
    <a
      className="text-6xl text-right ml-auto mr-6"
      href="/login"
      onClick={[Function]}
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
        <path
          d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zM7.35 18.5C8.66 17.56 10.26 17 12 17s3.34.56 4.65 1.5c-1.31.94-2.91 1.5-4.65 1.5s-3.34-.56-4.65-1.5zm10.79-1.38a9.947 9.947 0 00-12.28 0A7.957 7.957 0 014 12c0-4.42 3.58-8 8-8s8 3.58 8 8c0 1.95-.7 3.73-1.86 5.12z"
        />
        <path
          d="M12 6c-1.93 0-3.5 1.57-3.5 3.5S10.07 13 12 13s3.5-1.57 3.5-3.5S13.93 6 12 6zm0 5c-.83 0-1.5-.67-1.5-1.5S11.17 8 12 8s1.5.67 1.5 1.5S12.83 11 12 11z"
        />
      </svg>
    </a>
  </header>
  <div
    className="flex min-h-[85%]"
  >
    <div
      className="flex flex-col justify-top p-[2%] items-center space-y-8 w-[20%] bg-[#FBDCE2]"
    >
      <h1
        className="text-4xl font-bold mb-2"
      >
        Sort By 
      </h1>
      <div
        className="border-[#60AB9A] w-full h-1 border-2"
      />
      <div
        className="flex flex-row "
      >
        <div
          className="flex flex-col"
        >
          <button
            alt="handleDeadline"
            id="handleDeadline"
            onClick={[Function]}
          >
            <h1
              className="text-3xl m-2 font-bold mb-2"
            >
              Deadline 
            </h1>
            <div
              className="mr-2 border-[#60AB9A] w-full h-1 border-2"
            />
          </button>
        </div>
        <div
          className="flex flex-col"
        >
          <button
            alt="handlePriority"
            id="handlePriority"
            onClick={[Function]}
          >
            <h1
              className="text-3xl m-2 font-bold mb-2"
            >
              Priority 
            </h1>
            <div
              className=" ml-2 border-[#60AB9A] w-full h-1 border-2"
            />
          </button>
        </div>
      </div>
      <div
        className=" space-y-8 "
      >
        <div
          className="flex flex-col justify-center"
        >
          <h1
            className="text-4xl font-bold mb-2"
          >
            !!! Priority
          </h1>
          <div
            className="border-[#60AB9A] w-full h-1 border-2"
          />
          <button
            className="text-xl text-left font-bold mt-2 underline decoration-[#0acdff]"
            onClick={[Function]}
          >
            Task 1
          </button>
          <button
            className="text-xl text-left font-bold mt-2 underline decoration-[#0acdff]"
            onClick={[Function]}
          >
            Task 2
          </button>
          <button
            className="text-xl text-left font-bold mt-2 underline decoration-[#0acdff]"
            onClick={[Function]}
          >
            Task 3
          </button>
        </div>
        <div
          className="flex flex-col justify-center "
        >
          <h1
            className="text-4xl font-bold mb-2"
          >
            !! Priority
          </h1>
          <div
            className="border-[#60AB9A] w-full h-1 border-2"
          />
          <button
            className="text-xl text-left font-bold mt-2 underline decoration-[#0acdff]"
            onClick={[Function]}
          >
            Task 1
          </button>
          <button
            className="text-xl text-left font-bold mt-2 underline decoration-[#0acdff]"
            onClick={[Function]}
          >
            Task 2
          </button>
          <button
            className="text-xl text-left font-bold mt-2 underline decoration-[#0acdff]"
            onClick={[Function]}
          >
            Task 3
          </button>
        </div>
        <div
          className="flex flex-col justify-center"
        >
          <h1
            className="text-4xl font-bold mb-2"
          >
            ! Priority
          </h1>
          <div
            className="border-[#60AB9A] w-full h-1 border-2"
          />
          <button
            className="text-xl text-left font-bold mt-2 underline decoration-[#0acdff]"
            onClick={[Function]}
          >
            Task 1
          </button>
          <button
            className="text-xl text-left font-bold mt-2 underline decoration-[#0acdff]"
            onClick={[Function]}
          >
            Task 2
          </button>
          <button
            className="text-xl text-left font-bold mt-2 underline decoration-[#0acdff]"
            onClick={[Function]}
          >
            Task 3
          </button>
        </div>
      </div>
    </div>
    <div
      className="flex flex-col md:flex-row w-[80%]"
    >
      <div
        className="flex-1 flex-col p-[2%] justify-center items-center h-full"
      >
        <h1
          className="flex text-5xl font-bold mb-2"
        >
          Projects
        </h1>
        <div
          className="border-[#60AB9A] w-80 h-1 border-2"
        />
        <button
          className="flex text-2xl font-bold mt-2 underline decoration-[#0acdff]"
          onClick={[Function]}
        >
          Project 1
        </button>
        <button
          className="flex text-2xl font-bold mt-2 underline decoration-[#0acdff]"
          onClick={[Function]}
        >
          Project 2
        </button>
        <button
          className="flex text-2xl font-bold mt-2 underline decoration-[#0acdff]"
          onClick={[Function]}
        >
          Project 3
        </button>
        <button
          className="flex text-2xl font-bold mt-2 underline decoration-[#0acdff]"
          onClick={[Function]}
        >
          ...
        </button>
        <div
          className="my-16"
        />
        <h1
          className="flex text-5xl font-bold mb-2"
        >
          Overdue Tasks
        </h1>
        <div
          className="border-[#60AB9A] w-80 h-1 border-2"
        />
        <button
          className="flex text-2xl font-bold mt-2 underline decoration-[#0acdff]"
          onClick={[Function]}
        >
          Task 1
        </button>
        <button
          className="flex text-2xl font-bold mt-2 underline decoration-[#0acdff]"
          onClick={[Function]}
        >
          Task 2
        </button>
        <button
          className="flex text-2xl font-bold mt-2 underline decoration-[#0acdff]"
          onClick={[Function]}
        >
          Task 3
        </button>
        <button
          className="flex text-2xl font-bold mt-2 underline decoration-[#0acdff]"
          onClick={[Function]}
        >
          ...
        </button>
      </div>
      <div
        className="flex-1 flex-col p-[2%] justify-center items-center h-full"
      >
        <h1
          className="flex text-5xl font-bold mb-2"
        >
          Tasks
        </h1>
        <div
          className="border-[#60AB9A] w-80 h-1 border-2"
        />
        <button
          className="flex text-2xl font-bold mt-2 underline decoration-[#0acdff]"
          onClick={[Function]}
        >
          Task 1
        </button>
        <button
          className="flex text-2xl font-bold mt-2 underline decoration-[#0acdff]"
          onClick={[Function]}
        >
          Task 2
        </button>
        <button
          className="flex text-2xl font-bold mt-2 underline decoration-[#0acdff]"
          onClick={[Function]}
        >
          Task 3
        </button>
        <button
          className="flex text-2xl font-bold mt-2 underline decoration-[#0acdff]"
          onClick={[Function]}
        >
          Task 4
        </button>
        <button
          className="flex text-2xl font-bold mt-2 underline decoration-[#0acdff]"
          onClick={[Function]}
        >
          Task 5
        </button>
        <button
          className="flex text-2xl font-bold mt-2 underline decoration-[#0acdff]"
          onClick={[Function]}
        >
          Task 6
        </button>
        <button
          className="flex text-2xl font-bold mt-2 underline decoration-[#0acdff]"
          onClick={[Function]}
        >
          Task 7
        </button>
        <button
          className="flex text-2xl font-bold mt-2 underline decoration-[#0acdff]"
          onClick={[Function]}
        >
          Task 8
        </button>
        <button
          className="flex text-2xl font-bold mt-2 underline decoration-[#0acdff]"
          onClick={[Function]}
        >
          Task 9
        </button>
        <button
          className="flex text-2xl font-bold mt-2 underline decoration-[#0acdff]"
          onClick={[Function]}
        >
          ...
        </button>
      </div>
      <div
        className="my-16"
      />
    </div>
  </div>
</div>
`;
